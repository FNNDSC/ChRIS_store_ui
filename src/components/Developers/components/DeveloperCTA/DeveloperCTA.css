.developer-cta {
  /* color */
  background: #393f44;
  background: var(--pf-black-800);
  background: linear-gradient(135deg, #4d5258, #393f44);
  background: linear-gradient(135deg, var(--pf-black-700), var(--pf-black-800));
  background-image: url('../../../../assets/img/banner_chris.png');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;

  /* display */
  padding: 20px;
}

.developer-cta-overview, .developer-cta-form { margin: 15px; }

.developer-cta-overview { flex: 3; }

.developer-cta-form { flex: 2; }

.developer-cta-header, .developer-cta-desc {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);
}

.developer-cta-header {
  font-weight: 200;
  font-size: 3.4em;
  line-height: 1.3;
}

.developer-cta-desc {
  /* display */
  margin-top: 20px;

  /* font */
  font-weight: 600;
  font-size: medium;
  text-shadow: 3px 3px 6px #030303;
  text-shadow: 3px 3px 6px var(--pf-black);
}

/* responsive breakpoints */
@media (min-width: 480px)  {
  .developer-cta { padding: 4em; }
}
