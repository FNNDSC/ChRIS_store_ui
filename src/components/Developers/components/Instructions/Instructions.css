.instructions {
  /* display */
  padding: 20px;
}

.instructions-container {
  /* display */
  width: 100%;
  max-width: 1000px;
  margin: 0 auto !important;
}

.instructions-header { margin-bottom: 25px; }

.instructions-source {
  /* font */
  font-size: 0.5em;

  /* other */
  float: right;
}

.instructions-steps {
  /* border */
  border-top-color: #d1d1d1;
  border-top: 1px solid var(--pf-black-300); 
}

.instructions-step .instructions-number {
  /* font */
  font-size: 2em;
  font-weight: 500;
  line-height: 1.5;
}

.instructions-body {
  /* display */
  display: flex;

  /* flex */
  flex-flow: column;
  justify-content: center;
}

.instructions-body h3 {
  /* display */
  margin: 10px 0;

  /* font */
  line-height: 1.5;
}

.instructions-body .bash-line-container { margin: inherit; }

.instructions-body > *:first-child { margin-top: 0; }

.instructions-body > *:last-child { margin-bottom: 0; }

.instructions-step > * {
  /* display */
  margin: 30px 0;

  /* font */
  line-height: 2;
}

.instructions-step pre {
  /* color */
  background: #ededed;
  background: var(--pf-black-200);

  /* display */
  display: inline;
  padding: 0.1em 0.3em;

  /* font */
  font-size: inherit;

  /* border */
  border: 0;
}

.instructions-step:not(:last-child) {
  /* border */
  border-bottom-color: #d1d1d1;
  border-bottom: 1px solid var(--pf-black-300);
}

.instructions-number > sub {
  /* color */
  color: #8b8d8f;
  color: var(--pf-black-500);

  /* display */
  display: block;

  /* font */
  line-height: 1.3;
}

.instructions-body > a {
  /* display */
  display: block;
  margin: 15px 0;

  /* font */
  font-size: medium;
}

/* responsive breakpoints */
@media (min-width: 480px)  {
  .instructions { padding: 4em; }
}

@media (min-width: 700px) {
  .instructions-step { display: flex; }

  .instructions-number {
    /* flex */
    flex: 1;
  }

  .instructions-body {
    /* flex */
    flex: 3;
  }
}
