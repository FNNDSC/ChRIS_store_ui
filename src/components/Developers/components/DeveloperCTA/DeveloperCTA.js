import React from 'react';
import { Card, CardBody } from 'patternfly-react';
import './DeveloperCTA.css';
import DeveloperSignup from '../DeveloperSignup/DeveloperSignup';

const DeveloperCTA = () => (
  <div className="developer-cta">
    <div className="row developer-cta-container">
      <div className="developer-cta-overview">
        <div className="developer-cta-header">
          Expand the reach of your image processing software
        </div>
        <div className="developer-cta-desc">
          <p>
          ChRIS is an <strong>open source platform</strong> for medical
          analytics in the cloud, democratizing the development of image
          processing apps within an ecosystem following
            <strong>
              {' '}common standards, rather than disparate silos
            </strong>.
          </p>

          <p>
          A ChRIS Developer account enables you to submit your image
          processing application as a containerized ChRIS plugin and
          share it with the broader ChRIS community of researchers and
          clinicians. <strong>Join us!</strong>
          </p>
        </div>
      </div>
      <div className="developer-cta-form">
        <Card>
          <CardBody>
            <DeveloperSignup />
          </CardBody>
        </Card>
      </div>
    </div>
  </div>
);

export default DeveloperCTA;
