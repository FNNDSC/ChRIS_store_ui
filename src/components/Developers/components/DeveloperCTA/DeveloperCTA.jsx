import React from 'react';
import PropTypes from 'prop-types';

import './DeveloperCTA.css';

import { Card, CardBody, Grid, GridItem, Bullseye } from '@patternfly/react-core';
import ConnectedDeveloperSignup from '../DeveloperSignup/DeveloperSignup';
import ChrisStore from '../../../../store/ChrisStore';

export const DeveloperCTA = ({ store }) => (
  <Bullseye id="developer-cta">
    <Grid style={{ maxWidth: '1200px' }} hasGutter>
      <GridItem lg={6} xs={12}>
        <div id="developer-cta-header">
          <h2>Expand the reach of your image processing software</h2>
          <p style={{ fontSize: 'medium' }}>
            A ChRIS Developer account enables you to submit your image
            processing application as a containerized ChRIS plugin and
            share it with the broader ChRIS community of researchers and
            clinicians.
          </p>
        </div>
      </GridItem>
      <GridItem lg={1} xs={12}/>
      <GridItem lg={5} xs={12}>
        {
          store.get('isLoggedIn') ? null :
            <div id="developer-cta-form">
              <Card>
                <CardBody>
                  <h2 style={{ margin: '0.25em 0 1.5em 0' }}><b>Create a ChRIS Developer Account</b></h2>
                  <ConnectedDeveloperSignup />
                </CardBody>
              </Card>
            </div>
        }
      </GridItem>
    </Grid>
  </Bullseye>
);

export default ChrisStore.withStore(DeveloperCTA);

DeveloperCTA.propTypes = {
  store: PropTypes.objectOf(PropTypes.object).isRequired,
};
