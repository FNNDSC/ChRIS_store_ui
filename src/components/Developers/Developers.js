import React from 'react';
import DeveloperCTA from './components/DeveloperCTA/DeveloperCTA';
import Instructions from './components/Instructions/Instructions';

const Developers = () => (
  <div>
    <DeveloperCTA />
    <Instructions />
  </div>
);

export default Developers;
