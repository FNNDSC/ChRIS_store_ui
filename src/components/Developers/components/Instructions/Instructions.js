import React from 'react';
import './Instructions.css';
import BashLine from '../BashLine/BashLine';

const Instructions = () => (
  <div className="instructions">
    <div className="row no-flex">
      <h1 className="instructions-header">
        Get Started - 4 Simple Steps{' '}
        <a className="instructions-source" href="http://bit.ly/2KghHdY">[source]</a>
      </h1>
      <div className="instructions-steps">
        <div className="instructions-step">
          <div className="instructions-number">
            Step 1
            <sub>Setting up the environment</sub>
          </div>
          <div className="instructions-body">
            <h3>
              Create a <strong>Python3</strong> virtual environment for
              your plugin apps and activate it.
            </h3>
            <h3>
              Install the latest Cookiecutter in <pre>chrisapp_env</pre> if you haven{"'"}t installed it yet:
            </h3>
            <BashLine command="pip install -U cookiecutter" />
            <h3>
              Generate a ChRIS plugin app project:
            </h3>
            <BashLine command="cookiecutter https://github.com/FNNDSC/cookiecutter-chrisapp.git" />
            <h3>
              <strong>More information:</strong>
              <ul>
                <li>
                  <a
                    href="http://bit.ly/2Iih52m"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Creating the python virtual environment.
                  </a>
                </li>
                <li>
                  <a
                    href="http://bit.ly/2KghHdY"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Generating a ChRIS plugin project.
                  </a>
                </li>
              </ul>
            </h3>
          </div>
        </div>
        <div className="instructions-step">
          <div className="instructions-number">
            Step 2
            <sub>Creating a git repository</sub>
          </div>
          <div className="instructions-body">
            <h3>
              Create a new repository on{' '}
              <a href="https://github.com">https://github.com</a>
              <strong>/your_github_username</strong> with the same name as the
              app project directory generated by the previous cookiecutter
              command.
            </h3>
            <h3>
              Make the repository public.
            </h3>
            <h3>
              Don’t initialize the repository with a README, a license or
              a .gitignore file.
            </h3>
            <h3>
              Change the current working directory to the app project and
              initialize this local directory as a Git repository:
            </h3>
            <BashLine command="git init" />
          </div>
        </div>
        <div className="instructions-step">
          <div className="instructions-number">
            Step 3
            <sub>Push to git repository</sub>
          </div>
          <div className="instructions-body">
            <h3>
              Add and commit the files in the local repository:
            </h3>
            <BashLine command="git add ." />
            <BashLine
              command='git commit -m "First commit"'
              style={{ marginTop: 0 }}
            />
            <h3>
              Add the URL for the remote Github repository created
              in <pre>step 2</pre> where your local repository will be pushed:
            </h3>
            <BashLine command="git remote add origin **remote_Github_repository_URL** (eg. https://github.com/FNNDSC/pl-neuproseg.git)" />
            <BashLine command="git remote -v" style={{ marginTop: 0 }} />
            <h3>
              Push the changes in your local repository to GitHub:
            </h3>
            <BashLine command="git push origin master" />
          </div>
        </div>
        <div className="instructions-step">
          <div className="instructions-number">
            Step 4
            <sub>Add to Docker Hub</sub>
          </div>
          <div className="instructions-body">
            <h3>
              Create a new automated build and repository on your{' '}
              <a href="https://dockr.ly/2K2pnRF">Docker Hub</a> account.
            </h3>
            <h3>
              Once you log in, click the Create button in the header and select
              Automated Build from the drop-down menu. The website will walk you through
              setting up the automated build.
            </h3>
            <h3>
              Next, when prompted for the GitHub repository that you’d like to
              use for the automated build select the repository that you
              just created.
            </h3>
            <br />
            <h3>
              For more information on Automated Builds, visit the{' '}
              <a href="https://dockr.ly/2tmNDDz">Docker build documentation</a>.
            </h3>
            <h3>
              Modify <pre>requirements.txt</pre>, <pre>Dockerfile</pre> and the
              Python code with the proper versions of Python dependencies and
              libraries and push your changes to Github.
            </h3>
            <h3>
              Look at this{' '}
              <a href="http://bit.ly/2yzbRzF">simple <strong>fs</strong> plugin</a>
              {' '}or this{' '}
              <a href="http://bit.ly/2KbHosS">simple <strong>ds</strong> plugin</a>
              {' '}for guidance on getting started with your ChRIS plugin.
            </h3>
          </div>
        </div>
        <div className="instructions-step">
          <div className="instructions-number">
            Conclusion
            <sub>Extra resources</sub>
          </div>
          <div className="instructions-body">
            <h3>
              Once you{"'"}ve developed and properly tested your plugin app
              consult the <a href="http://bit.ly/2ltTJ0w">wiki</a> to learn how to register it to
              ChRIS.
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default Instructions;
