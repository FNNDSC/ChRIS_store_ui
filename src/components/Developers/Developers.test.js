import React from 'react';
import { shallow } from 'enzyme';
import Developers from './Developers';
import DeveloperCTA from './components/DeveloperCTA/DeveloperCTA';
import Instructions from './components/Instructions/Instructions';

describe('Developers', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Developers />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the DeveloperCTA component', () => {
    expect(wrapper.containsMatchingElement(<DeveloperCTA />)).toEqual(true);
  });

  it('should render the Instructions component', () => {
    expect(wrapper.containsMatchingElement(<Instructions />)).toEqual(true);
  });
});
