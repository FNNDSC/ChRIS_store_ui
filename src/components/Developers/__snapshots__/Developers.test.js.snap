// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developers should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Developers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DeveloperCTA />,
        <Instructions />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DeveloperCTA />,
          <Instructions />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
